### ISSUE TITLE
non-English attribute names + Rest API results in undesired behaviour

### ISSUE THREAD

**Issue by WillBrubaker** [woocommerce/woocommerce#13650]
## EXPLANATION OF THE ISSUE
In ticket: 543571-zd-woothemes ticket submitter reports that creating products via the API and then updating said products results in attribute values being wiped. This is reproducible and appears to be related to attribute names that are non-English (tested against Hebrew & Arabic with the same result on both)

## STEPS TO REPRODUCE THE ISSUE
1. Two global attributes exist. They are named "שָׁבוּר" and "مكسورة"
2. Each of these attributes has a term named "broken"
3. Create a product which has both of those attributes with an API call similar to this attached JSON 
[543571.json.zip](https://github.com/woocommerce/woocommerce/files/851170/543571.json.zip)
adjusting the attribute ids as appropriate.
4. Visit the product page - attribute values are present as expected.
5. Press the product's "Update" button, attribute values disappear.

## SYSTEM STATUS REPORT

```
`
### WordPress Environment ###

Home URL: https://c8tastrophe.com/wootest
Site URL: https://c8tastrophe.com/wootest
WC Version: 2.6.14
Log Directory Writable: ✔
WP Version: 4.7.3
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache/2.4.18 (Ubuntu)
PHP Version: 7.0.15-0ubuntu0.16.04.4
PHP Post Max Size: 8 MB
PHP Time Limit: 60
PHP Max Input Vars: 1000
cURL Version: 7.47.0
OpenSSL/1.0.2g

SUHOSIN Installed: –
Max Upload Size: 8 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 2.6.14
: 
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔
MaxMind GeoIP Database: ✔

### Active Plugins (1) ###

WooCommerce: by WooThemes – 2.6.14

### Settings ###

Force SSL: –
Currency: GBP (£)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2

### API ###

API Enabled: ✔

### WC Pages ###

Shop Base: #4 - /shop/
Cart: #5 - /cart/
Checkout: #6 - /checkout/
My Account: #7 - /my-account/

### Taxonomies ###

Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)


### Theme ###

Name: Twenty Seventeen
Version: 1.1
Author URL: https://wordpress.org/
Child Theme: ❌ – If you're modifying WooCommerce on a parent theme you didn't build personally
then we recommend using a child theme. See: How to create a child theme

WooCommerce Support: ✔

### Templates ###

Overrides: –
`
```

**Comment by mikejolley** [comment]
@claudiosanches another that needs confirming.

**Comment by Simonsbs** [comment]
Hey, I was the guy who reported the bug... so I'm just here to follow :)
It would be great to see this fixed soon since I use the API as the main source for adding products to my site, and it would be great if i could "tweak" products without having to go through the API.

**Comment by claudiosanches** [comment]
@WillBrubaker @Simonsbs @mikejolley I just tested it using REST API v1 and REST API v2 (using the JSON from here) on 3.0.0 RC 1 and works just fine, nothing is missing.

> Visit the product page - attribute values are present as expected.

Product page on frontend or backend?
Both for me don't have any issue.

> Press the product's "Update" button, attribute values disappear.

Tried product's "Update" button and the "Save attributes" button.

If we had some problem like it, probably already fixed in 3.0.0.
